kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -290,8 +412,121 @@ private static boolean hasPhysicalName(StructField field) {
     }
 
     private static int getColumnId(StructField field) {
-        // TODO: add a method to FieldMetadata to extract this as an int
-        // instead of having to parse it here
-        return Integer.parseInt(field.getMetadata().get(COLUMN_MAPPING_ID_KEY).toString());
+        return field.getMetadata().getLong(COLUMN_MAPPING_ID_KEY).intValue();
+    }
+
+    private static boolean hasNestedColumnIds(StructField field) {
+        return field.getMetadata().contains(COLUMN_MAPPING_NESTED_IDS_KEY);
+    }
+
+    private static FieldMetadata getNestedColumnIds(StructField field) {
+        return field.getMetadata().getMetadata(COLUMN_MAPPING_NESTED_IDS_KEY);
+    }
+
+    private static int getMaxNestedColumnId(StructField field) {
+        return getNestedColumnIds(field).getEntries().values().stream()
+                .filter(v -> v instanceof Long)
+                .map(v -> (Long) v)
+                .max(Long::compare)
+                .orElse(0L)
+                .intValue();
+    }
+
+    private static StructField transformSchema(
```

Comment:
javadocs with examples

Reply:
include docs for the params, what is the path? what does a path of "" represent? and why does it mean we have to go and calculate the physical name?

Reply:
added some javadoc that explains this